api/server.go

```diff
@@ -17,11 +16,10 @@ import (
 
 // api keeps track of the load-test API server state.
 type api struct {
-	agents       map[string]*loadtest.LoadTester
-	coordinators map[string]*coordinator.Coordinator
-	metrics      *performance.Metrics
-	coordLog     *mlog.Logger
-	agentLog     *mlog.Logger
+	resources sync.Map
```

Comment:
Today is the the day my old friend. 🤗

Reply:
Ah, I just did it to discuss a bit with @agnivade about adding two mutexes instead :stuck_out_tongue_closed_eyes:

Reply:
Lol.. If it's just out of laziness, then I would go for 2 separate mutexes. If there's indeed a performance improvement, then this is fine. Simplest way would be to write a benchmark and look at the results.

Reply:
I don't believe laziness had a part on this but neither did performance. 
This API has, most of the time, a single user and resources don't get created/destroyed very often so I don't believe optimizing for performance is that important.

My reasoning was to try and keep things clean by unifying the two resources and removing the need for the lock/unlock pattern. What I really didn't like was the idea that having a new type of resource meant adding yet another mutex and the way the underlying APIs are designed a single mutex was not an option.

Reply:
Ah I see now, it's the same issue as the gossip_client. I am wondering why a single mutex is not an option? Any reason why we cannot follow the approach in gossip_client?

```
	requestMut sync.RWMutex
	requestMap map[string]interface{}
```

We are already doing the casting with `sync.Map`.

Reply:
I admit I haven't tested this but my concern was about running a coordinator and agent on the same server (which is what we do now with the first load-test agent instance). Since the coordinator directly uses this API I feared this could cause a deadlock.
For example, stopping a coordinator through the API would mean acquiring the single lock to access the coordinator which in turn will make an HTTP request to stop the active agent. This will cause to acquire the lock again (the same lock).

Reply:
Hmm .. but the critical section is just the map access part right? If the sync.Map API does not deadlock, I am failing to see how can a mutex over a `map[string]interface{}` will. Perhaps I am missing something here?

For ease of use, we can even wrap the calls with Get/Set wrappers to make it equivalent to sync.Map calls.

Reply:
Yes, keeping the critical section to the map access would work fine. 
Somehow I was thinking of the usual lock/defer unlock pattern which would have caused problems.
I'll send an update to see how it looks.